Add language code prefixing to json translator

euwest opened the pull request:
fixes https://github.com/yola/gulp-static-i18n/issues/13

passing the `urlKeys` option(an array of keys at which urls are stored on json) will cause urls at those keys to be prefixed with a language code(e.g. `/es/`, `/zh-cn/` et al.) upon translation.
- add language code prefixing feature
- add unit/integration tests

beck commented:
This looks good.  The one thing you're missing is locale > lang conversation for the prefixing.

A bit of critique about your commits:
Do not commit your tests & functional changes separately.

Ideally each commit contains:
- tests
- app changes
- documentation

It is good practice to commit code often and push every so often to protect yourself when your machine rises up against you, but before a pull happens, rebase it and make it pretty.  Make each commit "matter" and have isolated purpose.

For this specific pull I would expect to see two commits:
1. Cleanup array handling
2. Add language code prefixing to json translator

beck commented on lib/translator.js:
```diff
@@ -136,7 +136,7 @@ Translator.prototype.translate = function(file) {
     var translatedFile;
     opts.lang = lang;
-    opts.translation = translator(copy, gettext);
+    opts.translation = translator(copy, gettext, lang);
```
this is passing a locale / gettext catalog name and not a language code.

euwest commented:
> The one thing you're missing is locale > lang conversation for the prefixing.

are you talking about `pt-br => pt_BR | zh-cn => zh_CN`? since this is public, it seems odd to handle our very specific language-code issues here. i guess we could include support for a `langMap` option that would look something like: 

``` js
var langMap = {
  "en": "",
  "pt-br": "pt_BR",
  "zh-cn": "zh_CN"
  ...
};
```

beck commented:
No no, write something [like this](https://github.com/django/django/blob/efa67b897b6ed5c6bbee1aa2646f4ba7ea6e2bc2/django/utils/translation/trans_real.py#L91)

beck commented:
And keep english prefixing

euwest committed:
clean up type checking, key fetching

euwest committed:
add language code prefixing to json translator
- add tests for new feature
- extend translators/json.js to prefix urlKeys with language codes

beck commented on test/test-sample-app.js:
```diff
@@ -61,9 +61,13 @@ describe('Static translation of an app with two locale directories', function()
+    expect(content).to.contain('"title":"Bonjour tout le monde"');
   });
 
+  it('prefixes the url in the json data with french language code', function() {
```
can you use Portuguese here? Then we'll get some test coverage on the lang prefixing

euwest committed:
switch test to portuguese for better coverage

beck commented:
:+1:
